homeassistant/components/metoffice/sensor.py

```
@@ -221,11 +221,7 @@ def native_value(self) -> StateType:
         elif self.entity_description.key == "weather" and hasattr(
             self.coordinator.data.now, self.entity_description.key
         ):
-            value = [
-                k
-                for k, v in CONDITION_CLASSES.items()
-                if self.coordinator.data.now.weather.value in v
-            ][0]
+            value = CONDITION_MAP.get(self.coordinator.data.now.weather.value)
```

This would previously fail hard